airflow/ui/openapi-gen/requests/types.gen.ts

```diff
@@ -96,6 +96,15 @@ export type NextRunAssetsResponse = {
   [key: string]: unknown;
 };
 
+export type HistoricalMetricsDataData = {
+  endDate: string;
+  startDate: string;
+};
+
+export type HistoricalMetricsDataResponse = {
+  [key: string]: unknown;
```

Comment:
@pierrejeambrun Is there a way we can better interpret the response type here?

In `v1-generated` it is just `object` which leads to `unknown` over here.

Reply:
Good question.

Indeed, most of the `private/ui/object` endpoints were only used by the UI and therefore did not have proper schema definition (for documentation purpose) and are just returning a plain 'dict' without any information which end up in the spec as a generic 'object'. This isn't ideal.

To better leverage the type system we could simply define a PydanticModel that defines the return type of the endpoint, and therefore doc and codegen would be accurate. I think we need to do that for all of the UI endpoints that we are migrating to have a cleaner interface. That will remove the need for manually duplicating types in the front-end like we used to do:
```
interface HistoricalMetricsData {
  dagRunStates: {
    [K in CamelCase<RunState>]: number;
  };
  dagRunTypes: {
    [K in CamelCase<DagRun["runType"]>]: number;
  };
  taskInstanceStates: {
    [K in TaskState extends string ? CamelCase<K> : never]: number;
  };
}
```

Reply:
Yes, that was a main motivation for AIP-84 in the first place, we didn't have good typing for any of our UI endpoints and manually writing them in the UI could break very easily.